core/capabilities/triggers/logevent/store.go

```diff
@@ -0,0 +1,71 @@
+package logevent
+
+import (
+	"fmt"
+	"sync"
+)
+
+type RegisterCapabilityFn[T any, Resp any] func() (*T, chan Resp, error)
+
+// Interface of the capabilities store
+type CapabilitiesStore[T any, Resp any] interface {
```

Comment:
looks like this is just a sync.Map with typing?  would it be simpler just to use sync.Map and cast (I am really surprised there is no generic version of sync.Map in core libraries but there you go)

Reply:
sync.Map does not have a InsertIfNotExists method. If we use LoadOrStore, we would still need to create a Trigger object with the underlying ContractReader object and pass it in as value. 

```
actual, _ = m.LoadOrStore(key, newValue)
```

To avoid creating a trigger object and have a concurrency safe solution, we'll have to use another sync.Mutex. We'll end up with a similar solution